censys/cli.py

```diff
@@ -0,0 +1,685 @@
+        format (str): What format to write the results. CSV, JSON, or stdout.
+        start_page (int): What page the query should start from.
+        max_pages (int): Adjust the number of results returned by the API.
+        api_secret (str): A API secret provided by Censys.
+        api_id (str): A API ID provided by Censys.
+    """
+
+    csv_fields: Fields = list()
+    """A list of fields to be used by the CSV writer."""
+
+    def __init__(self, **kwargs):
+        self.api_user = kwargs.get("api_id")
+        self.api_pass = kwargs.get("api_secret")
+
+        self.output_format = kwargs.get("format", "json")
+        self.start_page = kwargs.get("start_page", 1)
+        self.max_pages = kwargs.get("max_pages", 10)
+
+    @staticmethod
+    def _write_csv(filename: str, search_results: Results, fields: Fields) -> bool:
+        """
+        This method writes the search results to a new file in CSV format.
+
+        Args:
+            filename (str): The name of the file to write to on the disk.
+            search_results (Results): A list of search results from an API query.
+            fields (Fields): A list of fields to write as headers.
+
+        Returns:
+            bool: True if wrote to file successfully.
+        """
+
+        with open(filename, "w") as output_file:
+            if search_results and isinstance(search_results, list):
+                # Get the header row from the first result
+                writer = csv.DictWriter(output_file, fieldnames=fields)
+                writer.writeheader()
+
+                for result in search_results:
+                    # Use the Dict writer to process and write results to CSV
+                    writer.writerow(result)
+
+        print("Wrote results to file {}".format(filename))
+
+        # method returns True, if the file has been written successfully.
+        return True
+
+    @staticmethod
+    def _write_json(filename: str, search_results: Results) -> bool:
+        """
+        This method writes the search results to a new file in JSON format.
+
+        Args:
+            filename (str): name of the file to write to on the disk.
+            search_results (Results): list of search results from API query.
+
+        Returns:
+            bool: True if wrote to file successfully.
+        """
+
+        with open(filename, "w") as output_file:
+            # Since the results are already in JSON, just write them to a file.
+            json.dump(search_results, output_file, indent=4)
+
+        print("Wrote results to file {}".format(filename))
+        return True
+
+    @staticmethod
+    def _write_screen(search_results: Results) -> bool:
+        """
+        This method writes the search results to screen.
+
+        Args:
+            search_results (Results): list of search results from API query.
+
+        Returns:
+            bool: True if wrote to file successfully.
+        """
+
+        print(json.dumps(search_results, indent=4))
```

Comment:
Was going to suggest a JSON output for dumping to screen (ex: pipe through `jq`) but it looks like that's what "screen" already does :)

Reply:
Ah ya, I have been debating between `json.dumps` and `pprint`. Any thoughts?

Reply:
`pprint` might be more appropriate for 'screen' output. Having an option to optionally indicate target file - perhaps w/ `-` indicating output to stdout might be useful to facilitate chaining tools.

Reply:
Okay I will make `--format` and `--output` as well as switch to `pprint`

Reply:
Resolved in 1a182cc